OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -98,5 +98,17 @@ public static void NotifyBlocker(this Actor self, IEnumerable<CPos> positions)
 		{
 			NotifyBlocker(self, positions.SelectMany(p => self.World.ActorMap.GetUnitsAt(p)));
 		}
+
+		/// <summary>Get the `TerrainTypeInfo` for the cell containing this actor's CenterPosition.</summary>
+		public static TerrainTypeInfo TerrainTypeInfo(this Actor self)
```

Comment:
Actors (specifically buildings) may cover multiple terrain types, and we already have a large mess of terrain type querying methods, so I don't think this would be better left ungeneralized.